src/main/java/org/prebid/server/bidder/aja/AjaBidder.java

```diff
@@ -0,0 +1,188 @@
+package org.prebid.server.bidder.aja;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpMethod;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.aja.proto.ExtImpAja;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.EncodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class AjaBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpAja>> AJA_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpAja>>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public AjaBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<BidRequest>> result = new ArrayList<>();
+
+        final List<String> tagIds = new ArrayList<>();
+        final Map<String, Imp> impsByTagID = new HashMap<>();
+
+        for (Imp imp : bidRequest.getImp()) {
+            final ExtImpAja extImpAja = parseExtAJA(imp, errors);
+            if (extImpAja == null) {
+                continue;
+            }
+            imp = imp.toBuilder()
+                    .tagid(extImpAja.getAdSpotID())
+                    .ext(null)
+                    .build();
+
+            final String tagId = imp.getTagid();
+            if (!impsByTagID.containsKey(tagId)) {
```

Comment:
Probably you can remove this 'if' completely and later use impsByTagID.keySet(), the result should be the same.